experimental/ConsoleApp6/ConsoleApp6/Examples.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+
+// Example library that creates some simple metrics with a few default labels
+namespace SquidLibrary
+{
+    class SquidAnalyzer
+    {
+        static readonly MetricSource metrics = new MetricSource("SquidLibrary",
+            defaultLabels: new Dictionary<string, string>()
+            {
+                { "SquidBetaBuild", "true" },
+                { "SeafoodFeatureLevel", "4" }
+            });
+        static readonly Counter squidsAnalyzedCounter = metrics.CreateCounter("squids-analyzed");
```

Comment:
I'm wondering if there's an easy way (preferably at compile time) to get the caller's fully qualified class name.  That can act as the "default" name for this counter.  It may be done with reflections (but it's slow).

Reply:
Just like ILogger<ClassCategory>, we can also use generics to take a class name that we use for library name and we can get library version via reflection.